web/client/components/widgets/enhancers/dependenciesToWidget.js

```diff
@@ -12,6 +12,23 @@ const {pick} = require('lodash');
  *
  * @param {object} widget dependenciesMap. a map of key1: key2, where key1 is the key to return and key2 is the key of the value to return in the deps object
  * @param {object} deps the original dependencies object
+ *
+ * in case of nested dependencies
+ * @example
+ * // w1 dependenciesMap
+ * const dependenciesMap={
+ *   layer: "widgets[w1].layer",
+ *   mapSync: "widgets[w1].mapSync",
+ *   dependenciesMap: "widgets[w1].dependenciesMap"
+ * };
+ * const dependencies={
+ *   "widgets[w1].dependenciesMap": "widgets[w2].dependenciesMap",
+ *   "widgets[w1].layer": "widgets[w2].layer",
+ *   "widgets[w1].mapSync": true,
+ *   "widgets[w2].layer": {...},
+ * };
+ * return <EnhancedCmp dependenciesMap={dependenciesMap} dependencies={dependencies} />
+ * // the enhancer will pass to the component of dependencies={layer: {...}}
```

Comment:
not clear english. Didn't unserstand the mechanism. Something like: 

> Depedencies can be also transitive: in this case the in this case the transitive dependency will be expressed using ... (explain what the algorithm want to do)
example: one widget has layer --> widgets[w2]... --> widgets[w3]... dependencies will look like ... and dependenciesMap...